advanced_alchemy/extensions/litestar/exception_handler.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from litestar.connection import Request
+from litestar.connection.base import AuthT, StateT, UserT
+from litestar.exceptions import (
+    HTTPException,
+    InternalServerException,
+)
+from litestar.exceptions.responses import (
+    create_debug_response,  # pyright: ignore[reportUnknownVariableType]
+    create_exception_response,  # pyright: ignore[reportUnknownVariableType]
+)
+from litestar.response import Response
+from litestar.status_codes import (
+    HTTP_404_NOT_FOUND,
+    HTTP_409_CONFLICT,
+)
+
+from advanced_alchemy.exceptions import (
+    DuplicateKeyError,
+    ForeignKeyError,
+    IntegrityError,
+    NotFoundError,
+    RepositoryError,
+)
+
+if TYPE_CHECKING:
+    from litestar.connection import Request
+    from litestar.connection.base import AuthT, StateT, UserT
+    from litestar.response import Response
+
+
+class _HTTPConflictException(HTTPException):
```

Comment:
should this be inherited from `ClientException` so that is is consistent with how things are done in litestar?